eng/build.ps1

```diff
@@ -299,7 +299,7 @@ if ($BuildNodeJS) { $dotnetBuildArguments += "/p:BuildNodeJS=true" }
 # "-Projects ... -NoBuildJava -NoBuildManaged -NoBuildNodeJS". An empty `./build.ps1` command will build both
 # managed and native projects.
 $performDesktopBuild = ($BuildInstallers -and $Architecture -ne "arm") -or `
-    ($BuildNative -and -not $Architecture.StartsWith("arm", [System.StringComparison]::OrdinalIgnoreCase))
+    ($BuildNative -and -not $Architecture.Equals("arm", [System.StringComparison]::OrdinalIgnoreCase))
```

Comment:
@dougbu do you have any context on what this logic about desktop build / dotnetbuild is doing?  This tweak makes the arm64 leg fail with a whole bunch of restore errors, and metadata warnings.  But since we need the arm64 leg to actually produce arm64 dlls now

Reply:
I think this goes back to the conversation in the other PR, where the arm64 bits seem to be built in the x64 leg, as I see the following in the green x64 windows build leg:

```
  CommonLib.vcxproj -> D:\a\_work\1\s\artifacts\bin\CommonLib\x64\Release\CommonLib.lib
  CommonLib.vcxproj -> D:\a\_work\1\s\artifacts\bin\CommonLib\Win32\Release\CommonLib.lib
  Compiling Event Messages ...
  MC: Compiling D:\a\_work\1\s\src\Servers\IIS\AspNetCoreModuleV2\CommonLib\aspnetcore_msg.mc
  CommonLib.vcxproj -> D:\a\_work\1\s\artifacts\bin\CommonLib\arm64\Release\CommonLib.lib
```

Can we just build x64/win32/arm64 all in the main windows build job?

Reply:
> do you have any context on what this logic about desktop build / dotnetbuild is doing?

This logic is intended to avoid useless native builds on Windows ARM and ARM64. We previously needed desktop `msbuild` for the installer build on Windows ARM64 but nothing for the `$BuildNative` build on either platform. What you're doing could be fine, depending on how you fix up the targets, et cetera.

> Can we just build x64/win32/arm64 all in the main windows build job?

Yes and no (or maybe). The main Windows job can build do all of the `$BuildNative` stuff just fine. But eliminating the Windows ARM64 build entirely i.e. including the ARM64 installer build in the main Windows build job may lengthen that significantly. Try out internal builds and see.

Reply:
```suggestion
    ($BuildNative -and $Architecture -ne "arm")
```

or simplify the condition to

``` powershell
$performDesktopBuild = ($BuildInstallers -or $BuildNative) -and $Architecture -ne "arm"
```

(Note `-and` and `-or` unfortunately have equal precedence but, as you'd expect, `-ne` has higher precedence.)

Reply:
So this sounds like something I can backout and leave for the cleanup / developer PR too then since things were fine without this change